pandas/tests/indexes/test_multi.py

```diff
@@ -415,6 +415,16 @@ def test_set_value_keeps_names(self):
         self.assertIsNone(df.is_copy)
         self.assertEqual(df.index.names, ('Name', 'Number'))
 
+    def test_copy_names(self):
+        # Check that adding a "names" parameter to the copy is honored
+        # GH14302
+        multi_idx = pd.Index([(1, 2), (3, 4)], names=['MyName1', 'MyName2'])
+        multi_idx2 = multi_idx.copy(names=['NewName1', 'NewName2'])
+
+        self.assertTrue(multi_idx.equals(multi_idx2))
```

Comment:
Should we add `name` to `MultiIndex` as part of this patch then? What would the semantics mean then? We can't just duplicate the user's input *n* times since that would be a naming conflict.